pkgs/sdk/server/src/Internal/DataSources/PollingDataSource.cs

```diff
@@ -134,11 +136,17 @@ private void Dispose(bool disposing)
         {
             if (disposing)
             {
-                _canceller?.Cancel();
-                _featureRequestor.Dispose();
+                Shutdown(null);
             }
         }
 
+        private void Shutdown(DataSourceStatus.ErrorInfo? errorInfo)
+        {
+            _canceller?.Cancel();
+            _featureRequestor.Dispose();
+            _dataSourceUpdates.UpdateStatus(DataSourceState.Off, errorInfo);
```

Comment:
Emits events during dispose feels like something we shouldn't be doing. Because the thing we are emitting to could already be disposed/finalized.